### ISSUE TITLE
🐛 Panic when processing Django-style HTML syntax

### ISSUE THREAD

**Issue by brianhelba** [biomejs/biome#8605]
### Environment information

```block
CLI:
  Version:                      2.3.10
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v25.2.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.6.4

Biome Configuration:
  Status:                       Not set
  Path:                         unset

Workspace:
  Open Documents:               0
```

### What happened?

Here's a simple example repo to reproduce this bug: https://github.com/brianhelba/biome-panic

---
The output from this is:
```
> biome check

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: /__w/biome/biome/crates/biome_rowan/src/ast/mod.rs:206:13
Thread Name: biome::workspace_worker_0
Message: Tried to cast node with kind HTML_BOGUS as `"biome_html_syntax::generated::nodes::HtmlRoot"` but was unable to cast

 at src/django.html internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: Tried to cast node with kind HTML_BOGUS as `"biome_html_syntax::generated::nodes::HtmlRoot"` but was unable to cast
  
  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.
```

### Expected result

Biome should not panic, even when parsing unsupported (Django template syntax) files.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by brianhelba** [comment]
@dyc3 This follows the comment https://github.com/biomejs/biome/issues/8363#issuecomment-3690359597

**Comment by Netail** [comment]
Good to know: Angular also has this kind of interpolation syntax

**Comment by colecrouter** [comment]
Svelte appears to be suffering from the same issue. I couldn't pin down what was wrong in my project, until I saw this issue. Sure enough, Svelte syntax also features `{{`:

```svelte
<div
	transition:fly|global={{
		y: 20,
		duration: 200,
	}}
>
```

FWIW, this only happens with `html.experimentalFullSupportEnabled = true`, but I don't think that's relevant to the underlying issue.

**Comment by ematipico** [comment]
@colecrouter this is a different bug. Please open a new issue
